src/animation/fluidResistance.ts

```diff
@@ -0,0 +1,135 @@
+import { rAF } from '../rAF';
+import {
+  Entity,
+  applyForce,
+  fluidResistanceForceV,
+  gE,
+  getFluidPositionAtTerminalVelocity,
+} from '../forces';
+import { vector as Vector, addf, multf } from '../core';
+import { Listener, AnimationParams, AnimationInitializer } from './types';
+
+export interface FluidResistance1DParams extends AnimationParams {
+  config: {
+    mass: number;
+    rho: number;
+    area: number;
+    cDrag: number;
+    settle?: boolean;
+  };
+}
+
+interface FluidResistanceState {
+  mover: Entity;
+}
+
+/**
+ * A function to apply the drag force on each step in the course
+ * of the animation. First, we derive the force vector exerted by the
+ * fluid on the mover. Then we apply that vector to the mover to determine
+ * its next acceleration, velocity, and position.
+ */
+const applyFluidResistanceForceForStep = (
+  { mover }: FluidResistanceState,
+  config: FluidResistance1DParams['config']
+) => {
+  const dragForce = fluidResistanceForceV({
+    rho: config.rho,
+    area: config.area,
+    velocity: mover.velocity,
+    cDrag: config.cDrag,
+  });
+
+  const gravitationalForce: Vector<number> = [0, mover.mass * gE];
+  const force = multf({
+    v: addf({ v1: dragForce, v2: gravitationalForce }),
+    s: 0.001,
+  });
+
+  return applyForce({ force, entity: mover });
+};
+
+/**
+ * The fluid resistance function. This function tracks the internal state of the
+ * mover and starts the frame loop to apply the drag force as it moves.
+ */
+export const fluidResistance1D = ({
+  config,
+  onUpdate,
+  onComplete,
+}: FluidResistance1DParams): { controller: AnimationInitializer } => {
+  const state: FluidResistanceState = {
+    mover: {
+      mass: config.mass,
+      acceleration: [0, 0],
+      velocity: [0, 0],
+      position: [0, 0],
+    },
+  };
+
+  const tvPosition = getFluidPositionAtTerminalVelocity(config);
```

Comment:
The position at which our moving object will achieve terminal velocity. We pre-calculate this and use it as the stopping condition for the animation.